### PR TITLE
enable all inference and train on Gaudi/Gaudi2 with optimized perf with latest base

### PR DESCRIPTION
**Pull request by LeoZhao-Habana**

### CODE REVIEW

File: fastfold/habana/inject_habana.py [hunk 1]
```diff
@@ -49,9 +49,9 @@ def copy_qkv_linear(model_fast, ori_q, ori_k, ori_v):
 
 
 def copy_attention(model_fast, model_ori):
-    copy_linear(model_fast.to_q, model_ori.linear_q)
-    copy_linear(model_fast.to_k, model_ori.linear_k)
-    copy_linear(model_fast.to_v, model_ori.linear_v)
+    # copy_linear(model_fast.to_q, model_ori.linear_q)
```
Comment (Shenggan) [inline fastfold/habana/inject_habana.py:52]: maybe need to copy qkv weight here. `copy_qkv_linear(model_fast.to_qkv, model_ori.linear_q, model_ori.linear_k, model_ori.linear_v)`

File: fastfold/model/nn/primitives.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
     _chunk_slice,
 )
 
+from fastfold.habana.fastnn.custom_op import fused_softmax, fused_softmax_bias
```
Comment (Shenggan) [inline fastfold/model/nn/primitives.py:32]: This line may break on non-habana environment, maybe need `habana.is_habana()`

File: fastfold/model/hub/alphafold.py [hunk 3]
```diff
@@ -41,6 +41,7 @@
     tensor_tree_map,
 )
 
+from habana.hpuhelper import *
```
Comment (Shenggan) [inline fastfold/model/hub/alphafold.py:44]: This line may break on non-habana environment, maybe need `habana.is_habana()`

File: fastfold/habana/inject_habana.py (hunk 1 above)
Comment (LeoZhao-Habana) [reply fastfold/habana/inject_habana.py:52]: will fix in updated patch

File: fastfold/model/nn/primitives.py (hunk 2 above)
Comment (LeoZhao-Habana) [reply fastfold/model/nn/primitives.py:32]: will fix in updated patch

File: fastfold/model/hub/alphafold.py (hunk 3 above)
Comment (LeoZhao-Habana) [reply fastfold/model/hub/alphafold.py:44]: will fix in updated patch

**Comment by KatarinaYuan** [comment]
Hi,
    Thanks for this amazing work! If possible, could you please share some benchmarking for the difference of the training and inference time on Inter Habana Cluster or not?

**Comment by LeoZhao-Habana** [comment]
https://www.hpc-ai.tech/blog/intel-habana   here is blog for some details. It could run on both Gaudi and Gaudi2 for train and inference, we would publish more details and benchmarks in the future.

**Pull request merged**